### PR TITLE
UserUpdate Mutation / Service

### LINKED ISSUE #267: Update User Mutation / Service

### PR DESCRIPTION
**Pull request by Knerio**
close #267

### CODE REVIEW

File: spec/services/users/update_service_spec.rb [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        context 'when user is trying to modify its own admin status' do
+          let(:current_user) { create(:user) }
+          let(:user) { current_user }
+          let(:params) do
+            { admin: true }
+          end
+
+          it { is_expected.not_to be_success }
+
+          it 'updates user' do
```
Comment (Taucher2003) [inline spec/services/users/update_service_spec.rb:60]:
```suggestion
          it 'does not update user' do
```

File: spec/services/users/update_service_spec.rb [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+      it_behaves_like 'does not update'
+    end
+  end
+
+  context 'when user and params are valid' do
+    let(:params) do
+      { username: generate(:username) }
+    end
+    let(:organization) { create(:organization) }
```
Comment (Taucher2003) [inline spec/services/users/update_service_spec.rb:38]:
```suggestion
```

File: spec/services/users/update_service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    let(:current_user) { create(:user) }
+
+    context 'when user tries to update admin status' do
+      subject(:service_response) { described_class.new(current_user, user, params).execute }
+
```
Comment (Taucher2003) [inline spec/services/users/update_service_spec.rb:43]:
```suggestion
```

File: spec/services/users/update_service_spec.rb [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      context 'when user is admin' do
+        let(:params) do
+          { admin: true }
+        end
+        let(:current_user) { create(:user, :admin) }
+        let(:user) { create(:user) }
+
+        context 'when user is trying to modify its own admin status' do
+          let(:current_user) { create(:user) }
```
Comment (Taucher2003) [inline spec/services/users/update_service_spec.rb:52]:
```suggestion
```

File: spec/services/users/update_service_spec.rb (hunk 3 above)
Comment (Knerio) [reply spec/services/users/update_service_spec.rb:43]: We shouldnt remove that, that would test an other functionality

**Review by Taucher2003 — APPROVED** [review]

**Pull request merged**